core/domain/user_services.py

```diff
@@ -103,6 +104,45 @@ def get_user_id_from_username(username):
         return user_model.id
 
 
+def get_multi_user_ids_from_usernames(usernames):
+    """Gets the user_ids for a given list of usernames.
+
+    Args:
+        usernames: list(str). Identifiable usernames to display in the UI.
+
+    Returns:
+        list(str|None). Return the list of user ids corresponding to given
+        usernames.
+        """
+    user_ids: List[str] = []
+
+    if len(usernames) == 0:
+        return user_ids
+
+    normalized_usernames = [
+        user_domain.UserSettings.normalize_username(username)
+        for username in usernames
+    ]
+
+    found_models = user_models.UserSettingsModel.query(
+        user_models.UserSettingsModel.normalized_username.IN(
+            normalized_usernames
+        )
+    ).fetch()
+
+    username_to_user_id_map = {}
+    for model in found_models:
+        username_to_user_id_map[model.normalized_username] = model.id
+
+    for username in normalized_usernames:
+        if username in username_to_user_id_map:
+            user_ids.append(username_to_user_id_map[username])
+        else:
+            user_ids.append(None)
```

Comment:
```suggestion
        user_ids.append(username_to_user_id_map.get(username)
```

Reply:
It might be possible to simplify this even more to just one list comprehension.

Reply:
Done!